api/src/main/java/org/openmrs/module/appointmentscheduling/api/db/hibernate/HibernateAppointmentDAO.java

```diff
@@ -101,7 +101,7 @@ public Appointment getLastAppointment(Patient patient) {
 			throw new APIException("fromDate can not be later than toDate");
 
 		else {
-			String stringQuery = "SELECT appointment FROM Appointment AS appointment WHERE appointment.voided = 0";
+			String stringQuery = "SELECT appointment FROM Appointment AS appointment WHERE appointment.voided = '0'";
```

Comment:
Would `false` work?

Reply:
Yes false will work. I just tested it on both MySQL and PostgreSQL. I have now changed '0' to false.

Reply:
Am still wondering why `0` failed but `false` worked.

Reply:
On PostgreSQL, 0 is numeric and does not converts to boolean false. But when it is '0' then it gets automatically converts to boolean false ( see [here](https://www.postgresql.org/docs/9.1/datatype-boolean.html)).

For MySQL, tinyint(1) is synonymous to boolean. So either you send 0/1 or true/false it will work.

Reply:
Oh i see! 😊